src/ansys/acp/core/_tree_objects/extrusion_guide.py

```diff
@@ -0,0 +1,163 @@
+from .enums import (
+    ExtrusionGuideType,
+    extrusion_guide_type_from_pb,
+    extrusion_guide_type_to_pb,
+    status_type_from_pb,
+)
+from .object_registry import register
+from .virtual_geometry import VirtualGeometry
+
+# Workaround: these imports are needed to make sphinx_autodoc_typehints understand
+# the inherited members of the Elemental- and NodalData classes.
+import numpy as np  # noqa: F401 isort:skip
+
+__all__ = [
+    "ExtrusionGuide",
+]
+
+
+def to_direction_tuple_from_1D_array_or_none(array: DoubleArray) -> tuple[float, ...]:
+    """Convert a 1D DoubleArray or None protobuf message to a tuple with length 3."""
+    # This is needed because the extrusion direction is None if the extrusion guide type
+    # is not `by_direction`. See Feature 1122546 in ADO.
+    if len(array.data) == 0:
+        return 0.0, 0.0, 0.0
+    if not len(array.shape) == 1:
+        raise RuntimeError(f"Cannot convert {len(array.shape)}-dimensional array to tuple!")
+    value = tuple(array.data)
+    if len(value) == 3:
+        return value
+    raise RuntimeError(f"Cannot convert {value} to a direction of length 3!")
+
+
+@mark_grpc_properties
+@register
+class ExtrusionGuide(CreatableTreeObject, IdTreeObject):
+    """Instantiate an Extrusion Guide of a Solid Model.
+
+    Parameters
+    ----------
+    name :
+        The name of the Oriented Selection Set.
+    active :
+        Inactive extrusion guides are not used in the solid model extrusion.
+    edge_set :
+        Edge Set along which the Extrusion Guide acts.
+    extrusion_guide_type :
+        Type of the extrusion such as by direction or by geometry.
+    cad_geometry  :
+        CAD geometry along which the extrusion guide runs.
+        Needed if the extrusion type is set to :attr:`ExtrusionGuideType.BY_GEOMETRY`.
+    direction :
+        Direction along which the extrusion guide runs. Need if
+        the extrusion type is set to :attr:`ExtrusionGuideType.BY_DIRECTION`.
+    radius :
+        Controls the sphere of influence for mesh morphing.
+    depth :
+        Defines the bias of the mesh morphing.
+    use_curvature_correction :
+        Apply a curvature correction during the solid model extrusion which results in
+        a smoother extruded surface. Under certain circumstances, deactivating
+        curvature correction can lead to better extrusion results.
+
+    """
+
+    __slots__: Iterable[str] = tuple()
+
+    _COLLECTION_LABEL = "extrusion_guides"
+    _OBJECT_INFO_TYPE = extrusion_guide_pb2.ObjectInfo
+    _CREATE_REQUEST_TYPE = extrusion_guide_pb2.CreateRequest
+    _SUPPORTED_SINCE = "25.1"
+
+    def __init__(
+        self,
+        *,
+        name: str = "ExtrusionGuide",
+        active: bool = True,
+        edge_set: EdgeSet | None = None,
+        extrusion_guide_type: ExtrusionGuideType = "by_direction",
+        cad_geometry: VirtualGeometry | None = None,
+        direction: tuple[float, float, float] = (0.0, 0.0, 0.0),
```

Comment:
Current behavior:

```python
extrusion_guide = solid_model.create_extrusion_guide(extrusion_guide_type='by_geometry')
extrusion_guide.direction = (1., 2., 3.)
extrusion_guide.extrusion_guide_type = 'by_direction'
```
-> direction is `(0, 0, 1)`

I think we should raise on the `direction` getter if `extrusion_guide_type` is `by_geometry`, and in the `setter` if it's `by_geometry` for any value _other_ than `(0, 0, 0)`.